Makefile

```diff
@@ -286,6 +357,14 @@ all: all-c-tests all-cpp-tests all-cc-tests
 INCLUDE = -I ./$(L3_INCDIR)
 INCLUDE += -I ./$(dir $<)
 
+# use += here, so that extra flags can be provided via the environment
+
+# By default, L3-logging runs on its own. To integrate L3 with LOC machinery,
+# to squirrel away the loc_t ID value, run: L3_LOC_ENABLED=1 make ...
+ifdef L3_LOC_ENABLED
+    CFLAGS += -DL3_LOC_ENABLED
+    CFLAGS += -DLOC_FILE_INDEX=LOC_$(subst .,_,$(subst -,_,$(notdir $<)))
```

Comment:
This line is the Real-Magik! behind LOC machinery, which allows compilation to succeed cleanly w/o any work from the user's side.